### PR TITLE
Display library value sets along with data criteria

### PR DESCRIPTION
**Pull request by lukoktonos**
Previously, value sets from multiple libraries weren't entirely being added to the 'Data Criteria' list of value sets on the measure page.  This change adds them to the page on a new section that shows which library they are from.  This section does not appear if no library value sets are loaded.

Added a new section to the measure page to display value sets from libraries.

Added fixture with multiple value set libraries for testing.

Updated measure_view_spec to have QDM/CQL sections, added unit tests.
Screenshot:
image

Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [X] This pull request describes why these changes were made.
- [X] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1089
- [x] JIRA ticket links to this PR
- [X] Code diff has been done and been reviewed
- [X] If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced.
- [X] Tests are included and test edge cases
- [X] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [X] Automated regression test(s) pass

If JIRA tests were used to supplement or replace automated tests:
- [X] JIRA test links: (N/A)
- [X] Justification for using JIRA tests: (N/A)
- [X] JIRA tests have been added to sprint (N/A)

**Reviewer 1:**

Name: @aelliott 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass - N/A
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree - N/A

**Reviewer 2:**

Name: @c-monkey 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass - N/A
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree - N/A

### CODE REVIEW

File: app/assets/javascripts/views/measure_value_sets.js.coffee [hunk 1]
```diff
@@ -5,6 +5,8 @@ class Thorax.Views.MeasureValueSets extends Thorax.Views.BonnieView
     @summaryValueSets = [] # array of {generic value set descriptor, oid, and code}
     @dataCriteria = new Thorax.Collections.ValueSetsCollection([], sorting: 'complex')  # all criteria that aren't supplemental criteria
```
Comment (c-monkey) [inline app/assets/javascripts/views/measure_value_sets.js.coffee]: the comment on this line has become out of date

File: spec/javascripts/views/measure_view_spec.js.coffee [hunk 2]
```diff
@@ -1,82 +1,108 @@
...
+    beforeEach ->
+      @cqlMeasure = new Thorax.Models.Measure getJSONFixture('measure_data/CQL/CMS107/CMS107v6.json'), parse: true
+      @cqlPatients = new Thorax.Collections.Patients getJSONFixture('records/CQL/CMS107/patients.json'), parse: true
+      @cqlMeasureValueSetsView = new Thorax.Views.MeasureValueSets(model: @cqlMeasure, measure: @cqlMeasure, patients: @cqlPatients)
+      @cqlMeasureValueSetsView.appendTo 'body'
+
+    afterEach ->
+      @cqlMeasureValueSetsView.remove()
+
+    it 'renders library value sets', ->
```
Comment (c-monkey) [inline spec/javascripts/views/measure_view_spec.js.coffee]: CMS107 v6 actually has overlapping value sets between data criteria in the main library and in an additional library. Therefore, I think your test should add assertions about these values being visible.

**Review by c-monkey — CHANGES_REQUESTED** [review]
I recognize this PR will change potentially for BONNIE-388, but the current state looks great. I have a couple minor change requests.

**Comment by c-monkey** [comment]
As of today, this PR addresses the issue in https://oncprojectracking.healthit.gov/support/browse/BONNIE-360 and all unit tests pass.

File: spec/javascripts/views/measure_view_spec.js.coffee (hunk 2 above)
Comment (lukoktonos) [reply spec/javascripts/views/measure_view_spec.js.coffee]: Adding this test directly based on the others didn't work, so I will have to investigate why.

File: app/assets/javascripts/views/measure_value_sets.js.coffee [hunk 3]
```diff
@@ -3,69 +3,73 @@ class Thorax.Views.MeasureValueSets extends Thorax.Views.BonnieView
...
+    else
+      version = ''
+      codeConcepts = []
+
+    codes = new Backbone.PageableCollection(@sortAndFilterCodes(codeConcepts), @pagination_options)
+    if version.match(/^Draft/)
+      version = "Draft"
+    [version, codes]
+
+  setSummaryValueSets: (valueSet, oid, cid, name, codes) ->
```
Comment (alexanderelliott121) [inline app/assets/javascripts/views/measure_value_sets.js.coffee]: This name is misleading, given that the function adds a single valueset if it doesn't already exist.

Suggest:
addSummaryValueSet

**Review by c-monkey — DISMISSED** [review]
- implemented as https://oncprojectracking.healthit.gov/support/browse/BONNIE-388?focusedCommentId=76118&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-76118
- https://oncprojectracking.healthit.gov/support/browse/BONNIE-360 resolved

**Review by alexanderelliott121 — CHANGES_REQUESTED** [review]
Change name of method setSummaryValueSets to better reflect its function (suggest addSummaryValueSet)

**Review by alexanderelliott121 — APPROVED** [review]

**Review by c-monkey — APPROVED** [review]
all tests still pass; BONNIE-360 still resolved

**Review by alexanderelliott121 — APPROVED** [review]

**Pull request merged**